### PR TITLE
tests: Ensure exec_host() consistently captures command output

### PR DESCRIPTION
**Pull request by BbolroC**
The `exec_host()` function often fails to capture the output of a given command because the node debugger pod is prematurely terminated. To address this issue, the function has been refactored to ensure consistent output capture by adjusting the `kubectl debug` process as follows:

- Keep the node debugger pod running
- Wait until the pod is fully ready
- Execute the command using `kubectl exec`
- Capture the output and terminate the pod

This PR refactors `exec_host()` to implement the above steps, improving its reliability. And it reverts #10156 as well.

Fixes: #10081

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

File: tests/integration/kubernetes/tests_common.sh [hunk 1]
```diff
@@ -112,18 +121,15 @@ exec_host() {
 	# [bats-exec-test:38] INFO: k8s configured to use runtimeclass
 	# bash: line 1: $'\r': command not found
 	# ```
-	output="$(kubectl debug -qi "node/${node}" --image=quay.io/bedrock/ubuntu:latest -- chroot /host bash -c "${command}" | tr -d '\r')"
+	local command_output
+	command_output=$(kubectl exec -q "${new_debugger_pod}" -- chroot /host bash -c "${command}" | tr -d '\r' 2>/dev/null)
+	local exit_code=$?
```
Comment (sprt) [inline tests/integration/kubernetes/tests_common.sh:126]: Have we verified that `kubectl exec` returns the underlying exit code?

**Review by sprt — COMMENTED** [review]
Thank you @BbolroC, this is much needed! I left a few comments/questions 🙂

File: tests/integration/kubernetes/tests_common.sh [hunk 2]
```diff
@@ -95,14 +95,23 @@ delete_pod_if_new() {
...
+	local old_debugger_pods=($(kubectl get pods -o name | grep node-debugger))
+
+	# Run a debug pod
+	kubectl debug -q "node/${node}" --image=quay.io/bedrock/ubuntu:latest -- chroot /host bash -c "sleep 5m" >&2
 
+	# Identify the new debugger pod
+	local new_debugger_pod=$(get_new_debugger_pod "${old_debugger_pods[@]}")
+
+	# Wait for the newly created pod to be ready
+	kubectl wait --timeout="30s" --for=condition=ready "${new_debugger_pod}" > /dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:112]: We need `> /dev/null` indeed as any stdout here could ruin the output of the function. But for error messages, I agree with removing `2>&1`. Thanks!

File: tests/integration/kubernetes/tests_common.sh [hunk 3]
```diff
@@ -95,14 +95,23 @@ delete_pod_if_new() {
...
 	command="${@:2}"
-	command+='; echo -en \\n$?'
 
-	# Get the already existing debugger pods.
-	declare -a old_debugger_pods=( $(kubectl get pods -o name | grep node-debugger) )
+	# Get the already existing debugger pods
+	local old_debugger_pods=($(kubectl get pods -o name | grep node-debugger))
+
+	# Run a debug pod
+	kubectl debug -q "node/${node}" --image=quay.io/bedrock/ubuntu:latest -- chroot /host bash -c "sleep 5m" >&2
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:106]: yeah, let's use `sleep infinity` then. I just copied and pasted Gaby's. 😉

File: tests/integration/kubernetes/tests_common.sh [hunk 4]
```diff
@@ -95,14 +95,23 @@ delete_pod_if_new() {
 #	$1 - the node name
 #
 exec_host() {
-	node="$1"
+	local node="$1"
 	# `kubectl debug` always returns 0, so we hack it to return the right exit code.
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:99]: I think I don't get your point here. Could you elaborate more on this? Thanks!

File: tests/integration/kubernetes/tests_common.sh (hunk 1 above)
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:126]: yeah, we could preserve the logic using `command+='; echo -en \\n$?'` for this. thanks.

File: tests/integration/kubernetes/tests_common.sh [hunk 5]
```diff
@@ -112,18 +121,15 @@ exec_host() {
 	# [bats-exec-test:38] INFO: k8s configured to use runtimeclass
 	# bash: line 1: $'\r': command not found
 	# ```
-	output="$(kubectl debug -qi "node/${node}" --image=quay.io/bedrock/ubuntu:latest -- chroot /host bash -c "${command}" | tr -d '\r')"
+	local command_output
+	command_output=$(kubectl exec -q "${new_debugger_pod}" -- chroot /host bash -c "${command}" | tr -d '\r' 2>/dev/null)
```
[the remark this answers is not in the document]
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:125]: yeah, we could remove the error redirection here if we want to see an error message. thanks!

File: tests/integration/kubernetes/tests_common.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (BbolroC) [reply tests/integration/kubernetes/tests_common.sh:99]: ah, removing the comment? I will do it. Thanks!

**Review by GabyCT — APPROVED** [review]

File: tests/integration/kubernetes/tests_common.sh [hunk 6]
```diff
@@ -95,14 +95,23 @@ delete_pod_if_new() {
 #	$1 - the node name
 #
 exec_host() {
-	node="$1"
-	# `kubectl debug` always returns 0, so we hack it to return the right exit code.
```
Comment (sprt) [inline tests/integration/kubernetes/tests_common.sh:99]: Let's reinstate this comment since we reintroduced this logic

**Review by sprt — APPROVED** [review]
LGTM but let's please readd that comment 🙂

**Review by danmihai1 — APPROVED** [review]
Thanks @BbolroC!

**Pull request merged**